llm/cpp/group_beam_searcher.hpp

```diff
@@ -234,7 +234,7 @@ struct GroupBeamSearcher {
             group->is_done(parameters);
             if (!group->done) {
                 for (const Beam& beam : group->ongoing) {
-                    next_tokens.push_back({beam.tokens.back(), beam.global_beam_idx});
+                    next_tokens.push_back({beam.tokens.back(), int32_t(beam.global_beam_idx)});
```

Comment:
Why do we need to transpose tokens/beams to pack it into `TokenToBeam`? Can we just return two vectors: tokens and corresponding beam global indices? Later in the code we can just use elements in vectors as a host memory for tensors `input_ids` and `beam_idx` without need to manually copy each element in the loop. It is negligible from the performance side, but the code would be simpler. What do you think?

Reply:
Updated